save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesController.kt

```diff
@@ -17,104 +27,211 @@ import org.springframework.transaction.annotation.Transactional
     fun getFilteredTestSuites(
         @RequestParam(required = false, defaultValue = "") tags: String,
         @RequestParam(required = false, defaultValue = "") name: String,
         @RequestParam(required = false, defaultValue = "") language: String,
+        @RequestParam(required = false, defaultValue = "false") isContest: Boolean,
         authentication: Authentication,
     ): Flux<TestSuiteDto> = Mono.just(TestSuiteFilters(name, language, tags))
         .flatMapMany {
-            testSuitesService.findTestSuitesMatchingFilters(it)
-        }
-        .filter {
-            testSuitePermissionEvaluator.canAccessTestSuite(it, authentication)
-        }
-        .map {
-            it.toDto(it.requiredId())
+            testSuitesService.findTestSuitesMatchingFilters(it).toFlux()
         }
+        .mapToDtoFiltered(authentication, isContest)
 
-    /**
-     * @param id id of the test suite
-     * @return response with test suite with provided id
-     */
     @GetMapping("/internal/testSuite/{id}")
-    fun getTestSuiteById(@PathVariable id: Long) =
+    @PreAuthorize("permitAll()")
+    @Operation(
+        method = "GET",
+        summary = "Get test suite by id.",
+        description = "Get test suite by id.",
+    )
+    @Tag(name = "internal")
+    @Parameters(
+        Parameter(name = "id", `in` = ParameterIn.PATH, description = "id of test suite", required = true),
+    )
+    @ApiResponse(responseCode = "200", description = "Successfully fetched filtered test suites.")
+    fun getTestSuiteById(@PathVariable id: Long): ResponseEntity<TestSuite?> =
             ResponseEntity.status(HttpStatus.OK).body(testSuitesService.findTestSuiteById(id))
 
-    /**
-     * Trigger update of standard test suites. Can be called only by superadmins externally.
-     *
-     * @return response entity
-     */
     @PostMapping(path = ["/api/$v1/updateStandardTestSuites", "/internal/updateStandardTestSuites"])
     @PreAuthorize("hasRole('ROLE_SUPER_ADMIN')")
-    fun updateStandardTestSuites() = Mono.just(quartzScheduler)
+    @Tags(
+        Tag(name = "superadmins"),
+        Tag(name = "internal"),
+    )
+    @Operation(
+        method = "POST",
+        summary = "Update standard test suites.",
+        description = "Trigger update of standard test suites. Can be called only by super admins externally.",
+    )
+    @ApiResponse(responseCode = "200", description = "Successfully updated standard test suites.")
+    fun updateStandardTestSuites(): Mono<Unit> = Mono.just(quartzScheduler)
         .map {
             it.triggerJob(
                 UpdateJob.jobKey
             )
         }
 
-    /**
-     * @param testSuiteDtos suites, which need to be deleted
-     * @return response entity
-     */
     @PostMapping("/internal/deleteTestSuite")
     @Transactional
-    fun deleteTestSuite(@RequestBody testSuiteDtos: List<TestSuiteDto>) =
+    @Tag(name = "internal")
```

Comment:
I thought we don't need to add swagger api for internal endpoints

Reply:
Let's unify endpoint documentation